src/styles/main.scss

```diff
@@ -1,3 +1,64 @@
+$cell-size: 100px;
+$gap-size: 1px;
+$max-width: 720px;
+$color-pink: #ffc;
+$color-grey: #eee;
+$start-day:
+  mon 0,
+  tue 1,
+  wed 2,
+  thu 3,
+  fri 4,
+  sut 5,
+  sun 6;
+
 body {
   margin: 0;
 }
+
+.calendar {
+  padding: 20px 10px;
+  list-style: none;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  width: $max-width;
+
+  @for $month from 1 through 31 {
+    & :nth-child(#{$month})::before {
+      content: "#{$month}";
+      font-family: Arial, Helvetica, sans-serif;
+      font-size: 30px;
+    }
+  }
+
+  @for $day from 28 through 31 {
+    &--month-length--#{$day} > :not(:nth-child(-n + #{$day})) {
+      display: none;
+    }
+  }
+
+  @each $day, $num in $start-day {
+    &--start-day--#{$day} > :first-child {
+      margin-left: calc((#{$cell-size + $gap-size}) * #{$num});
+    }
+  }
+
+  &__item {
+    box-sizing: border-box;
```

Comment:
It's the general rule that will be applied to all elements inside your project, will be better to specify it with `*` selector